python/torch_mlir_e2e_test/test_suite/index_put.py

```diff
@@ -818,3 +818,27 @@ def forward(self, input, index, value):
 def IndexPutHackedTwin3DIntAccumulateModule_basic(module, tu: TestUtils):
     module.forward(torch.randint(1000, (10, 8, 6)), torch.randint(4, (5, )),
                    torch.randint(1000, (5, 8, 6)))
+
+class IndexAddModule(torch.nn.Module):
+
+  def __init__(self):
+    super().__init__()
+
+  @export
+  @annotate_args([
+      None,
+      ([-1, -1, -1], torch.float32, True),
+      ([-1], torch.float32, True),
+      ([-1], torch.float32, True),
+      ([-1, -1], torch.float32, True),
+  ])
+  def forward(self, input, dim, index, value):
+    return torch.ops.aten.index_add(input, dim, index, value)
+
+
+@register_test_case(module_factory=lambda: IndexAddModule())
+def IndexAddModule_basic(module, tu: TestUtils):
+  module.forward(torch.rand((3, 5, 2), dtype=torch.float32),
+          torch.tensor(0.0),
+          torch.tensor([0, 1]),
```

Comment:
This is an int tensor, which does not match the dtype in the annotation

Reply:
Can you clarify what you mean here?

Reply:
The tensor has `dtype` int:

```python
In [3]: torch.tensor([0, 1]).dtype
Out[3]: torch.int64
```

which does not match the corresponding annotation here:

https://github.com/llvm/torch-mlir/blob/a8e50a68720ab37e832cc121224d5a8fbb8a78b9/python/torch_mlir_e2e_test/test_suite/index_put.py#L832